catroid/src/createatschool/java/org/catrobat/catroid/createatschool/ui/adapter/TemplateAdapter.java

```diff
@@ -33,37 +33,49 @@
 import android.widget.RelativeLayout;
 import android.widget.TextView;
 
+import com.google.common.base.Strings;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
 import com.squareup.picasso.Picasso;
 
 import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.Constants;
+import org.catrobat.catroid.common.TemplateContainer;
 import org.catrobat.catroid.common.TemplateData;
-import org.catrobat.catroid.createatschool.common.TemplateConstants;
+import org.catrobat.catroid.createatschool.transfers.FetchTemplatesTask;
 import org.catrobat.catroid.utils.TextSizeUtil;
+import org.catrobat.catroid.utils.ToastUtil;
+
+public class TemplateAdapter extends ArrayAdapter<TemplateData> implements
+		FetchTemplatesTask.OnFetchTemplatesCompleteListener {
+
+	private static LayoutInflater inflater;
 
-public class TemplateAdapter extends ArrayAdapter<TemplateData> {
 	private final Context context;
 	private OnTemplateEditListener onTemplateEditListener;
-	private static LayoutInflater inflater;
+	private TemplateContainer templateContainer;
 
 	public TemplateAdapter(Context context, int resource, int textViewResourceId) {
 		super(context, resource, textViewResourceId);
 		this.context = context;
 		inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
-		initTemplates();
+		fetchTemplates();
 	}
 
-	private void initTemplates() {
-		add(new TemplateData(TemplateConstants.TEMPLATE_ACTION_NAME, TemplateConstants.TEMPLATE_ACTION_IMAGE_NAME,
-				TemplateConstants.TEMPLATE_ACTION_LANDSCAPE_FILENAME, TemplateConstants.TEMPLATE_ACTION_PORTRAIT_FILENAME));
-
-		add(new TemplateData(TemplateConstants.TEMPLATE_ADVENTURE_NAME, TemplateConstants.TEMPLATE_ADVENTURE_IMAGE_NAME,
-				TemplateConstants.TEMPLATE_ADVENTURE_LANDSCAPE_FILENAME, TemplateConstants.TEMPLATE_ADVENTURE_PORTRAIT_FILENAME));
+	private void fetchTemplates() {
+		FetchTemplatesTask task = new FetchTemplatesTask(context);
+		task.setOnFetchTemplatesCompleteListener(this);
+		task.execute();
+	}
 
-		add(new TemplateData(TemplateConstants.TEMPLATE_PUZZLE_NAME, TemplateConstants.TEMPLATE_PUZZLE_IMAGE_NAME,
-				TemplateConstants.TEMPLATE_PUZZLE_LANDSCAPE_FILENAME, TemplateConstants.TEMPLATE_PUZZLE_PORTRAIT_FILENAME));
+	private void initTemplates() {
+		if (templateContainer == null) {
```

Comment:
It could be e.g. when there is an invalid server response and TemplateContainer class could not be deserialized.